### PR TITLE
small fixes in docs: bullet lists in transpiler levels

### PR DESCRIPTION
**Pull request by 1ucian0**
### Summary

This PR fixes the bullet list for the transpilation level in the documentation, besides some other small issues (like double "the"s)

Before 
Screenshot 2023-07-06 at 11 24 45

After
Screenshot 2023-07-06 at 11 24 21

### CODE REVIEW

File: qiskit/compiler/transpiler.py [hunk 1]
```diff
@@ -210,10 +210,12 @@ def transpile(
...
-            * 0: no optimization
-            * 1: light optimization
-            * 2: heavy optimization
-            * 3: even heavier optimization
+
+                * 0: no optimization
+                * 1: light optimization
+                * 2: heavy optimization
+                * 3: even heavier optimization
+
```
Comment (jakelishman) [inline qiskit/compiler/transpiler.py:218]: The indentation isn't needed for bullet lists, they just need a blank line before the first and after the last bullet.  This one was missing a blank before, which is probably why it wasn't rendering.

File: qiskit/execute_function.py [hunk 2]
```diff
@@ -148,10 +148,12 @@ def execute(
...
-            #. No optimization
-            #. Light optimization
-            #. Heavy optimization
-            #. Highest optimization
+
+                * No optimization
+                * Light optimization
+                * Heavy optimization
+                * Highest optimization
+
```
Comment (jakelishman) [inline qiskit/execute_function.py:156]: `#.` makes a numbered list, and I think this list is logically meant to look like "0. No optimisation; 1. Light optimisation; ...".  If so, I think this was intended:
```rst
... at the expend of longer transpilation time.

0. No optimization
#. Light optimization
#. Heavy optimization
#. Highest optimization
```

(`#.` autonumbers in rst)

Comment (1ucian0) [reply qiskit/execute_function.py]: oops.. yes, my intentation was to unified it with the previous style: 
```
              * 0: no optimization
              * 1: light optimization
              * 2: heavy optimization
              * 3: even heavier optimization
```

done in 915c7e96b59b0721344d0c8f7bee61df1211d099

File: qiskit/compiler/transpiler.py (hunk 1 above)
Comment (1ucian0) [reply qiskit/compiler/transpiler.py]: Indeed, fixed in 915c7e96b59b0721344d0c8f7bee61df1211d099

**Review by Cryoris — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**